core/drivers/scmi-msg/smt.c

```diff
@@ -164,12 +154,10 @@ static void scmi_proccess_smt(unsigned int agent_id, uint32_t *payload_buf)
 out:
 	if (error) {
 		DMSG("SCMI error");
-		smt_status |= SMT_STATUS_ERROR;
+		smt_hdr->status |= SMT_STATUS_ERROR | SMT_STATUS_FREE;
+	} else {
+		smt_hdr->status |= SMT_STATUS_FREE;
 	}
-
-	smt_status |= SMT_STATUS_FREE;
-
-	atomic_store_u32(&smt_hdr->status, smt_status);
```

Comment:
If I understood it correctly was the purpose with this atomic store is to make sure there's no partial updates since there might be concurrent readers. If that's the case then `atomic_store_*()` should be used or you'd need to play games with `volatile` which I wouldn't recommend.

Reply:
Checked back, but no, we don't expect concurrent readers, as readers readers have to get data from the buffer before allowing one to use the buffer for some other message. So in the end, current readers would be a programming error.

Yet, a single reader, notified by an interrupt, would test `status` cell in the buffer to check if buffer is updated with some message. For such cases I think those 2 bit should be updated in a single access.